File: generate_products.js
const fs = require('fs');

const categories = [
    { id: "electronics", subCategories: ["Mobile & Wearables", "Computing & Peripherals", "Home Entertainment", "Smart Home & Security"] },
    { id: "fashion", subCategories: ["Men's Fashion", "Women's Fashion", "Jewelry & Timepieces", "Footwear"] },
    { id: "home", subCategories: ["Kitchen & Dining", "Decor & Lighting", "Furniture & Bedding", "Garden & Outdoor"] },
    { id: "beauty", subCategories: ["Skincare", "Cosmetics", "Personal Care"] },
    { id: "health", subCategories: ["Health & Fitness", "Vitamins & Supplements", "Medical Supplies"] },
    { id: "kids", subCategories: ["Baby Essentials", "Toys & Learning", "Kids Clothing"] },
    { id: "grocery", subCategories: ["Grocery & Gourmet", "Snacks & Sweets", "Beverages"] },
    { id: "specialized", subCategories: ["Automotive Gear", "Pet Supplies", "Sports & Outdoors", "Arts & Hobbies"] },
    { id: "digital", subCategories: ["Learning & Software", "Digital Tools", "eBooks"] }
];

const brands = {
    "electronics": ["Sony", "Samsung", "Apple", "Dell", "HP", "Asus", "LG", "Anker", "Razer", "Logitech"],
    "fashion": ["Nike", "Adidas", "Levi's", "Puma", "Gucci", "Zara", "H&M", "Ray-Ban", "Rolex"],
    "home": ["IKEA", "Dyson", "Ninja", "Philips", "Breville", "KitchenAid", "Nespresso", "Cuisinart"],
    "beauty": ["MAC", "L'Oreal", "Maybelline", "Clinique", "Estee Lauder", "Dior", "Chanel", "The Ordinary"],
    "health": ["Optimum", "Fitbit", "Garmin", "GNC", "NatureMade", "Omron", "Theragun"],
    "kids": ["Lego", "Fisher-Price", "Mattel", "Hasbro", "Pampers", "Huggies", "Disney", "Nerf"],
    "grocery": ["Nestle", "Kelloggs", "Coca-Cola", "Pepsi", "Oreo", "Quaker", "Heinz", "Lavazza"],
    "specialized": ["Shimano", "KONG", "Coleman", "Yamaha", "Bosch", "Michelin", "Purina", "Pedigree"],
    "digital": ["Adobe", "Microsoft", "Udemy", "Coursera", "Notion", "Kaspersky", "TurboTax"]
};

const adjectives = ["Premium", "Ultra", "Elite", "Pro", "Advanced", "Classic", "Smart", "Compact", "Deluxe", "Essential", "Original", "Wireless"];
const nouns = {
    "electronics": ["Smartphone", "Laptop", "Monitor", "Headphones", "Speaker", "Camera", "Tablet", "Smartwatch"],
    "fashion": ["Jacket", "Sneakers", "T-Shirt", "Jeans", "Watch", "Sunglasses", "Handbag", "Dress"],
    "home": ["Blender", "Coffee Maker", "Vacuum", "Air Fryer", "Sofa", "Desk", "Lamp", "Rug"],
    "beauty": ["Lipstick", "Foundation", "Moisturizer", "Serum", "Perfume", "Shampoo", "Cleanser", "Mascara"],
    "health": ["Protein Powder", "Yoga Mat", "Treadmill", "Vitamins", "Dumbbells", "Foam Roller", "Blood Pressure Monitor"],
    "kids": ["Action Figure", "Board Game", "Building Blocks", "Stroller", "Diapers", "Plush Toy", "Puzzle"],
    "grocery": ["Coffee Beans", "Olive Oil", "Pasta", "Chocolate", "Tea", "Honey", "Almonds", "Cereal"],
    "specialized": ["Tent", "Bike Helmet", "Dog Food", "Car Wash Kit", "Dash Cam", "Fishing Rod", "Drill", "Guitar"],
    "digital": ["Software", "Course Bundle", "Subscription", "eBook", "App License", "Template Pack", "Antivirus"]
};

const badges = ["", "", "", "New", "Hot", "Sale", "Best Seller", "Popular", "Premium", "Trending", "Deal"];

function randomChoice(arr) { return arr[Math.floor(Math.random() * arr.length)]; }
function uniform(min, max) { return Math.random() * (max - min) + min; }

function generateProducts() {
    let products = [];
    let pidCounter = 1;

    for (let cat of categories) {
        let catId = cat.id;
        for (let i = 0; i < 55; i++) {
            let brand = randomChoice(brands[catId]);
            let adj = randomChoice(adjectives);
            let noun = randomChoice(nouns[catId]);

            let modelChar = randomChoice(['X', 'Pro', 'Max', 'Plus', 'Eco', 'Lite', `Series ${Math.floor(uniform(1, 8))}`]);

            let title = `${brand} ${adj} ${noun} ${modelChar}`;
            if (catId === "fashion") {
                title = `${brand} ${adj} ${noun}`;
            } else if (catId === "grocery") {
                title = `${adj} ${brand} ${noun} ${randomChoice(['Pack', 'Bundle', '1kg', '500g', 'Family Size'])}`;
            }

            let originalPrice = Math.round(uniform(10.0, 1500.0) * 100) / 100;
            if (["grocery", "beauty", "kids"].includes(catId)) {
                originalPrice = Math.round(uniform(5.0, 100.0) * 100) / 100;
            }

            let discountPct = randomChoice([0, 0, 0, 10, 15, 20, 30, 50]);
            let price = Math.round(originalPrice * (1 - discountPct / 100.0) * 100) / 100;
            if (price === originalPrice) {
                price = Number((originalPrice - 0.01).toFixed(2));
            }

            let rating = Math.round(uniform(4.0, 5.0) * 10) / 10;
            let reviews = Math.floor(uniform(50, 150000));

            let badge = randomChoice(badges);
            let subCat = randomChoice(cat.subCategories);

            let image = `https://picsum.photos/seed/${pidCounter * 10}/400/400`;

            products.append = products.push({
                id: `${catId.charAt(0)}${pidCounter}`,
                title,
                category: catId,
                subCategory: subCat,
                price,
                originalPrice,
                rating,
                reviews,
                image,
                badge
            });
            pidCounter++;
        }
    }
    return products;
}

const products = generateProducts();
const productsJsonStr = JSON.stringify(products, null, 4);

// Replace in script.js
const frontendPath = "frontend/script.js";
let fContent = fs.readFileSync(frontendPath, 'utf8');
const fPattern = /const PRODUCTS = \[\s*[\s\S]*?\n\s*\];/;
fContent = fContent.replace(fPattern, "const PRODUCTS = " + productsJsonStr + ";");
fs.writeFileSync(frontendPath, fContent);
console.log(`Updated ${frontendPath}`);

// Replace in products.js
const backendPath = "backend/routes/products.js";
let bContent = fs.readFileSync(backendPath, 'utf8');
const bPattern = /const DUMMY_PRODUCTS = \[[\s\S]*?\];/;
const bJsonStr = productsJsonStr.replace(/"id":/g, '"_id":');
bContent = bContent.replace(bPattern, "const DUMMY_PRODUCTS = " + bJsonStr + ";");
fs.writeFileSync(backendPath, bContent);
console.log(`Updated ${backendPath}`);
console.log(`Generated 495 products successfully!`);

File: generate_products.py
import os
import json
import random
import re

categories = [
    {"id": "electronics", "subCategories": ["Mobile & Wearables", "Computing & Peripherals", "Home Entertainment", "Smart Home & Security"]},
    {"id": "fashion", "subCategories": ["Men's Fashion", "Women's Fashion", "Jewelry & Timepieces", "Footwear"]},
    {"id": "home", "subCategories": ["Kitchen & Dining", "Decor & Lighting", "Furniture & Bedding", "Garden & Outdoor"]},
    {"id": "beauty", "subCategories": ["Skincare", "Cosmetics", "Personal Care"]},
    {"id": "health", "subCategories": ["Health & Fitness", "Vitamins & Supplements", "Medical Supplies"]},
    {"id": "kids", "subCategories": ["Baby Essentials", "Toys & Learning", "Kids Clothing"]},
    {"id": "grocery", "subCategories": ["Grocery & Gourmet", "Snacks & Sweets", "Beverages"]},
    {"id": "specialized", "subCategories": ["Automotive Gear", "Pet Supplies", "Sports & Outdoors", "Arts & Hobbies"]},
    {"id": "digital", "subCategories": ["Learning & Software", "Digital Tools", "eBooks"]}
]

brands = {
    "electronics": ["Sony", "Samsung", "Apple", "Dell", "HP", "Asus", "LG", "Anker", "Razer", "Logitech"],
    "fashion": ["Nike", "Adidas", "Levi's", "Puma", "Gucci", "Zara", "H&M", "Ray-Ban", "Rolex"],
    "home": ["IKEA", "Dyson", "Ninja", "Philips", "Breville", "KitchenAid", "Nespresso", "Cuisinart"],
    "beauty": ["MAC", "L'Oreal", "Maybelline", "Clinique", "Estee Lauder", "Dior", "Chanel", "The Ordinary"],
    "health": ["Optimum", "Fitbit", "Garmin", "GNC", "NatureMade", "Omron", "Theragun"],
    "kids": ["Lego", "Fisher-Price", "Mattel", "Hasbro", "Pampers", "Huggies", "Disney", "Nerf"],
    "grocery": ["Nestle", "Kelloggs", "Coca-Cola", "Pepsi", "Oreo", "Quaker", "Heinz", "Lavazza"],
    "specialized": ["Shimano", "KONG", "Coleman", "Yamaha", "Bosch", "Michelin", "Purina", "Pedigree"],
    "digital": ["Adobe", "Microsoft", "Udemy", "Coursera", "Notion", "Kaspersky", "TurboTax"]
}

adjectives = ["Premium", "Ultra", "Elite", "Pro", "Advanced", "Classic", "Smart", "Compact", "Deluxe", "Essential", "Original", "Wireless"]
nouns = {
    "electronics": ["Smartphone", "Laptop", "Monitor", "Headphones", "Speaker", "Camera", "Tablet", "Smartwatch"],
    "fashion": ["Jacket", "Sneakers", "T-Shirt", "Jeans", "Watch", "Sunglasses", "Handbag", "Dress"],
    "home": ["Blender", "Coffee Maker", "Vacuum", "Air Fryer", "Sofa", "Desk", "Lamp", "Rug"],
    "beauty": ["Lipstick", "Foundation", "Moisturizer", "Serum", "Perfume", "Shampoo", "Cleanser", "Mascara"],
    "health": ["Protein Powder", "Yoga Mat", "Treadmill", "Vitamins", "Dumbbells", "Foam Roller", "Blood Pressure Monitor"],
    "kids": ["Action Figure", "Board Game", "Building Blocks", "Stroller", "Diapers", "Plush Toy", "Puzzle"],
    "grocery": ["Coffee Beans", "Olive Oil", "Pasta", "Chocolate", "Tea", "Honey", "Almonds", "Cereal"],
    "specialized": ["Tent", "Bike Helmet", "Dog Food", "Car Wash Kit", "Dash Cam", "Fishing Rod", "Drill", "Guitar"],
    "digital": ["Software", "Course Bundle", "Subscription", "eBook", "App License", "Template Pack", "Antivirus"]
}

badges = ["", "", "", "New", "Hot", "Sale", "Best Seller", "Popular", "Premium", "Trending", "Deal"]

def generate_products():
    products = []
    pid_counter = 1
    
    for cat in categories:
        cat_id = cat["id"]
        for i in range(55): # 55 products per category
            brand = random.choice(brands[cat_id])
            adj = random.choice(adjectives)
            noun = random.choice(nouns[cat_id])
            
            # Additional details to make it realistic
            version = str(random.randint(1, 9))
            model_char = random.choice(['X', 'Pro', 'Max', 'Plus', 'Eco', 'Lite', 'Series ' + str(random.randint(1,7))])
            
            title = f"{brand} {adj} {noun} {model_char}"
            if cat_id == "fashion":
                title = f"{brand} {adj} {noun}"
            elif cat_id == "grocery":
                title = f"{adj} {brand} {noun} {random.choice(['Pack', 'Bundle', '1kg', '500g', 'Family Size'])}"
                
            original_price = round(random.uniform(10.0, 1500.0), 2)
            if cat_id in ["grocery", "beauty", "kids"]:
                original_price = round(random.uniform(5.0, 100.0), 2)
                
            discount_pct = random.choice([0, 0, 0, 10, 15, 20, 30, 50])
            price = round(original_price * (1 - discount_pct/100.0), 2)
            if price == original_price:
                price = original_price - 0.01 # ensure x.99 aesthetic sometimes
            
            # Give a high-rating bias
            rating = round(random.uniform(4.0, 5.0), 1)
            reviews = random.randint(50, 150000)
            
            badge = random.choice(badges)
            subCat = random.choice(cat["subCategories"])
            
            # Simple placeholder image based on category via Picsum
            image = f"https://picsum.photos/seed/{pid_counter * 10}/400/400"
            
            p = {
                "id": f"{cat_id[0]}{pid_counter}",
                "title": title,
                "category": cat_id,
                "subCategory": subCat,
                "price": price,
                "originalPrice": original_price,
                "rating": rating,
                "reviews": reviews,
                "image": image,
                "badge": badge
            }
            products.append(p)
            pid_counter += 1
            
    return products

products = generate_products()
products_json_str = json.dumps(products, indent=4)

frontend_path = r"e:\New folder (2)\frontend\script.js"
backend_path = r"e:\New folder (2)\backend\routes\products.js"

# Write Frontend
with open(frontend_path, "r", encoding="utf-8") as f:
    f_content = f.read()

# Replace `const PRODUCTS = [...];`
f_pattern = re.compile(r"const PRODUCTS = \[\s*//(.*?)\n\];", re.DOTALL)
# It might match differently depending on comments, let's broadly match the array and everything inside until the semicolon
f_pattern2 = re.compile(r"const PRODUCTS = \[[\s\S]*?\];", re.MULTILINE)

f_content_new = f_pattern2.sub("const PRODUCTS = " + products_json_str + ";", f_content)

with open(frontend_path, "w", encoding="utf-8") as f:
    f.write(f_content_new)
print(f"Updated {frontend_path}")

# Write Backend
with open(backend_path, "r", encoding="utf-8") as f:
    b_content = f.read()

b_pattern = re.compile(r"const DUMMY_PRODUCTS = \[[\s\S]*?\];", re.MULTILINE)
b_json_str = products_json_str.replace('"id":', '"_id":')
b_content_new = b_pattern.sub("const DUMMY_PRODUCTS = " + b_json_str + ";", b_content)

with open(backend_path, "w", encoding="utf-8") as f:
    f.write(b_content_new)
print(f"Updated {backend_path}")
print(f"Generated {len(products)} products total.")
